pkg/operator/ceph/controller/predicate.go

```diff
@@ -436,6 +436,29 @@ func WatchControllerPredicate() predicate.Funcs {
 				if isUpgrade {
 					return true
 				}
+
+			case *cephv1.CephCOSIDriver:
+				objNew := e.ObjectNew.(*cephv1.CephCOSIDriver)
+				logger.Debug("update event on CephCOSIDriver CR")
+				// If the labels "do_not_reconcile" is set on the object, let's not reconcile that request
+				IsDoNotReconcile := IsDoNotReconcile(objOld.GetLabels())
+				if IsDoNotReconcile {
+					logger.Debugf("object %q matched on update but %q label is set, doing nothing",
+						objNew.Name, DoNotReconcileLabelName)
+					return false
+				}
+				diff := cmp.Diff(objOld.Spec, objNew.Spec)
+				if diff != "" {
+					logger.Infof("CephCOSIDriver CR has changed for %q. diff=%s", objNew.Name, diff)
+					return true
+				} else if objectToBeDeleted(objOld, objNew) {
+					logger.Debugf("CephCOSIDriver CR %q is going be deleted", objNew.Name)
+					return true
+				} else if objOld.GetGeneration() != objNew.GetGeneration() {
+					logger.Debugf("skipping CephCOSIDriver resource %q update with unchanged spec", objNew.Name)
+				}
+				logger.Debugf("no change in CephCOSIDriver CR %q", objNew.Name)
+
```

Comment:
@travisn : do we need upgrade flow like other crds here?
```
// Handling upgrades
				isUpgrade := isUpgrade(objOld.GetLabels(), objNew.GetLabels())
				if isUpgrade {
					return true
				}

```

Reply:
TBH, I think the `isUpgrade()` code is vestigial and should be removed. However, it is probably best to add it to this predicate now and handle the vestigial removal in a separate PR. It seems better to keep things consistent just in case.


https://github.com/rook/rook/blob/b693a5cca5c6e5e06dedb2b9b4ae9404ef44205a/pkg/operator/ceph/controller/predicate.go#L654-L674